app/src/main/java/com/example/notesapp/data/local/note/model/NoteEntity.kt

```diff
@@ -1,14 +1,22 @@
-package com.example.notesapp.Models
+package com.example.notesapp.data.local.note.model
 
 import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.PrimaryKey
 
-
 @Entity(tableName = "notes_table")
-data class Note(
+data class NoteEntity(
     @PrimaryKey(autoGenerate = true) val id : Int?,
     @ColumnInfo(name = "title") val title : String?,
     @ColumnInfo(name = "note") val note : String?,
     @ColumnInfo(name = "date") val date : String?
 ) : java.io.Serializable
+
+fun NoteModel.toNoteEntity(): NoteEntity {
```

Comment:
What's the upside for making the change from "Note" to "NoteEntity"? I see this is one of the most significant changes

Reply:
Is it for null safety?

Reply:
We must have two different models, one of them for DataBase (Entity model) and other one for app interaction (Domain Model[NoteModel]), with this aproach you will remove unecesary RoomDB functions, it's important to keep separate Room DB entities and Domain Models

Reply:
Hahaha, I'm sorrry I forgot to answer the nullable question, with the nullable fields you have an option to not fill these variables, now I think that it's not necesary to have them, because if you want to save one note, you will have these content, but thinking about how Note Application works in some smartphones, you have an option to save one note without content (it sounds weird, but happens), and also save a note without title